### PR TITLE
Hide 1st section header in WS front pages when width <600px

### LINKED ISSUE #1977: Hide 1st Section header in WS frontpages
**Is your feature request related to a problem? Please describe.**
Following https://github.com/bbc/psammead/pull/610 we have support to hide a section header on viewports <600px

**Describe the solution you'd like**
Pass the required prop, to hide the section header, on the first story in the first section in the page.

This can likely be done by passing a `section number`, using the index key on the 2 `maps` used. Then when they equal `0` , we know its the the first section :)

**Testing notes**
No cypress tests required, visual check cross-browser is sufficient. Unit tests will cover this.

Dev insight: Unit tests

### PR DESCRIPTION
**Pull request by Bopchy**
Resolves #1977 

**Overall change:** _Updates FrontPageSection to use latest version of @bbc/psammead-section-label._

**Code changes:**

- _Add sectionNumber prop to FrontPageSection._
- _Use visuallyHidden prop in SectionLabel._
- _Update snapshots._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: src/app/containers/FrontPageMain/index.jsx [hunk 1]
```diff
@@ -32,7 +32,11 @@ const FrontPageMain = ({ frontPageData }) => {
         <GhostWrapper>
           <GridItemConstrainedLarge>
             {groups.map(group => (
```
Comment (dr3) [inline src/app/containers/FrontPageMain/index.jsx]: You can get the index from the map :)
```suggestion
             {groups.map((group, index) => (
```

File: src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -153,6 +153,7 @@ exports[`FrontPageSection Container snapshots should render correctly for canoni
         },
       }
     }
+    visuallyHidden={false}
```
Comment (dr3) [inline src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap]: This snapshot has `visuallyHidden={false}` 3 times, but no `visuallyHidden={true}`, howcome? I would expect the first to be `visuallyHidden={true}`

**Review by dr3 — CHANGES_REQUESTED** [review]
Just https://github.com/bbc/simorgh/pull/2021/files#r295824064 mainly, would like some test coverage for the `visualHidden = sectionNumber === 0`

File: src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (Bopchy) [reply src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap]: Hadn't specified the `sectionNumber`. `sectionNumber` is now specified as 0 for the test cases, as there is only one `group` in each of test data

File: src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap [hunk 3]
```diff
@@ -153,6 +153,7 @@ exports[`FrontPageSection Container snapshots should render correctly for canoni
         },
       }
     }
+    visuallyHidden={true}
```
Comment (dr3) [inline src/app/containers/FrontPageSection/__snapshots__/index.test.jsx.snap:156]: Sorry, now we have the opersite problem, theyre all true 🙈 Can we get a tests that shows false too please?

**Review by dr3 — CHANGES_REQUESTED** [review]
Can this PR also be blocked on https://github.com/bbc/psammead/pull/672 please?

File: src/app/containers/FrontPageSection/index.test.jsx [hunk 4]
```diff
@@ -153,17 +153,17 @@ describe('FrontPageSection Container', () => {
 
     shouldShallowMatchSnapshot(
       'should render correctly for canonical',
-      <FrontPageSection group={group} />,
+      <FrontPageSection group={group} sectionNumber={0} />,
```
Comment (andrew-nowak) [inline src/app/containers/FrontPageSection/index.test.jsx:157]: Sorry, I know you changed these to make them test when `sectionNumber === 0`, could we also have one extra snapshot test where `sectionNumber === 1`? (or your favourite non-zero number)

**Review by andrew-nowak — CHANGES_REQUESTED** [review]
Looking good! Just a small reqeust about the test cases

**Comment by dr3** [comment]
@Bopchy https://www.npmjs.com/package/@bbc/psammead-story-promo-list has now been updated, so this PR is now unblocked :) installing the new version should fix is gap issue :)

File: src/app/containers/FrontPageSection/index.test.jsx (hunk 4 above)
Comment (Bopchy) [reply src/app/containers/FrontPageSection/index.test.jsx:157]: One of the snapshots now has `sectionNumber === 1`

File: package.json [hunk 5]
```diff
@@ -91,7 +91,7 @@
     "@bbc/psammead-inline-link": "^0.3.4",
     "@bbc/psammead-media-indicator": "^0.2.1",
     "@bbc/psammead-paragraph": "^1.0.2",
-    "@bbc/psammead-section-label": "^0.1.0-alpha.0",
+    "@bbc/psammead-section-label": "^0.1.1",
     "@bbc/psammead-sitewide-links": "^0.3.9",
     "@bbc/psammead-story-promo": "^0.2.0-alpha.6",
     "@bbc/psammead-story-promo-list": "^0.1.0-alpha.2",
```
Comment (dr3) [inline package.json]: Can you bump story promo list here please?

**Review by dr3 — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (Bopchy) [reply package.json]: Updated it.

File: src/app/containers/FrontPageSection/index.test.jsx [hunk 6]
```diff
@@ -205,10 +205,44 @@ describe('FrontPageSection Container', () => {
...
+          <FrontPageSection group={group} sectionNumber={0} />
+        </ServiceContextProvider>,
+      );
+
+      global.innerWidth = 500;
+      global.dispatchEvent(new Event('resize'));
+      const sectionLabel = container.getElementsByTagName('section')[0]
+        .firstChild;
+
+      expect(sectionLabel).toHaveStyleRule('clip-path', 'inset(100%)', {
```
Comment (dr3) [inline src/app/containers/FrontPageSection/index.test.jsx]: Hey, soz I dont think we should have tests for the styles. Instead we should assert that `SectionLabel` was called with `visuallyHidden` true or false

**Review by dr3 — CHANGES_REQUESTED** [review]
Just 1 thing about the tests, other code LGTM, looks good locally

**Review by dr3 — APPROVED** [review]

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

**Comment by david-boydell** [comment]
Looks good, this is ready for merge

**Review by andrew-nowak — APPROVED** [review]

**Pull request merged**